proposals/lip-0053.md

```diff
@@ -252,178 +479,192 @@ crossChainUpdateTransactionParams = {
 }
 ```
 
-In the following, we use the notation:
-
-* let `CCU` be the cross-chain update transaction.
-* let `inboxUpdate = CCU.params.inboxUpdate`.
-* let `partnerChainID = CCU.params.sendingChainID`.
-* let `partnerChain = chainAccount(partnerChainID)`.
-* let `partnerChannel = channel(partnerChainID)`.
-* let `partnerValidators = validators(partnerChainID)`.
-
-### Parameters Validity
-
-#### Liveness of Partner Chain
-
-A CCU transaction is only valid if the partner chain is not terminated and follows the [liveness rule][lip-0045#livenessCondition]. This is done by asserting the two points below:  
-
-* `partnerChain.status != CHAIN_TERMINATED`,
-* if `partnerChain.status == CHAIN_ACTIVE` then also validate `isLive(partnerChainID) == True`. The `isLive` function is specified in [LIP 0045][lip-0045].
-
-##### Liveness Requirement for the First CCU
-
-If `partnerChain.status == CHAIN_REGISTERED`, all proposed CCUs must contain a non-empty `certificate` which must follow the schema defined in  [LIP 0061][lip-0061]. In the following, let `certificate` be the deserialized certificate.
-
-Furthermore, if the CCU also contains a non-empty `inboxUpdate`, the certificate is only valid if it allows the sidechain account to remain live for a reasonable amount of time. This is done by checking that:
+##### Verification
 
 ```python
-timestamp - CCU.params.certificate.timestamp < LIVENESS_LIMIT / 2
+def verify(trs: Transaction) -> None:
+    # Liveness condition is only checked on the mainchain.
+    verifyLivenessCondition(trs)
+    if len(trs.params.certificate) > 0:
+        verifyCertificate(trs)
+    if len(trs.params.activeValidatorsUpdate) > 0 or trs.params.newCertificateThreshold > 0:
```

Comment:
If I am not mistaken, a relayer could always leave the fields `trs.params.activeValidatorsUpdate` and `trs.params.newCertificateThreshold` empty. Then the validators hash of the last certificate may be updated, but the corresponding validators are not.

Reply:
We changed the protocol to avoid sending an empty `certificateThreshold`. Now `validators(trs.params.sendingChainID).certificateThreshold` must always be set. I am not sure however that I understand your point.

Reply:
I think the logic is no all fine, I resolve this comment.